### PR TITLE
[batch] Use sops for credentials

### PR DESCRIPTION
**Pull request by lgruen**
Context: https://hail.zulipchat.com/#narrow/stream/300487-Hail-Batch-Dev/topic/ci_config.20secrets

#assign services

### CODE REVIEW

**Comment by lgruen** [comment]
Friendly ping, as merging this would help keep our fork in sync.

**Comment by jigold** [comment]
@daniel-goldstein I think I am fine with this change in theory. However, it is unclear to me how this would work if we added our encoded files to the repository. Would other forks have overwrite ours each time they merged changes? Would you want me to follow the new directions in my hail-jigold test project to verify the new changes are good?

**Comment by daniel-goldstein** [comment]
Forks would indeed need to overwrite ours, but since the file wouldn't change much it seems like that's not much of a hassle to maintain, right Leo? And ya this seems like a fine change but we would need to follow up right afterward with our own credentials.

**Comment by lgruen** [comment]
> I think if it's not too hard of a change, we should add the files with encoded secrets to something like `infra/gcp/data/...`. This makes it clear that these files have a different purpose and gives some indication that they're specific to your repo. If you want to also add prefixing the file name with the repo, then that would make it even clearer. But if it's too much work, don't bother. Maybe something like `infra/gcp/data/hail-is/` etc.

I've put all deployment-specific configs in an `$ORGANIZATION_DOMAIN` subfolder now, which hopefully should avoid any collisions.

**Review by jigold — DISMISSED** [review]

File: infra/gcp/README.md [hunk 1]
```diff
@@ -49,15 +40,14 @@ Instructions:
...
 
-- Install terraform.
+- Create `infra/gcp/$ORGANIZATION_DOMAIN/global.tfvars` based on the template below, where `$ORGANIZATION_DOMAIN` corresponds to the `organization_domain` setting in the template (e.g. `hail.is`). This avoids collisions between configuration files from different Hail deployments.
 
-- Create `$HOME/.hail/global.tfvars` that looks like:
 
    ```
    # organization_domain is a string that is the domain of the organization
-   # E.g. "broadinstitute.org"
+   # E.g. "hail.is"
```
[the remark this answers is not in the document]
Comment (lgruen) [reply infra/gcp/README.md:50]: Yes, that sounds fine too! It's another variable to explain the semantics of (as it doesn't appear in `ci_config.json`), but it actually clarifies the `<REPO>` reference at the end of the README. I'll push a commit to change this.

**Comment by lgruen** [comment]
Just realized that this is still open. Are there any missing changes?

**Comment by jigold** [comment]
I'm so sorry. I didn't see you addressed my comments about the domain. I'll double check it one last time, but I'm sure it's fine.

**Review by jigold — CHANGES_REQUESTED** [review]

File: infra/gcp/README.md [hunk 2]
```diff
@@ -42,13 +42,16 @@ Instructions:
 
   Download the client secret as `/tmp/auth_oauth2_client_secret.json`.
 
-- Create `infra/gcp/$ORGANIZATION_DOMAIN/global.tfvars` based on the template below, where `$ORGANIZATION_DOMAIN` corresponds to the `organization_domain` setting in the template (e.g. `hail.is`). This avoids collisions between configuration files from different Hail deployments.
+- Create `infra/gcp/$GITHUB_ORGANIZATION/global.tfvars` based on the template below, where `$GITHUB_ORGANIZATION` corresponds to the GitHub organization used for your Hail Batch deployment (e.g. [`hail-is`](https://github.com/hail-is/hail)). This avoids collisions between configuration files from different Hail deployments.
```
[the remark this answers is not in the document]
Comment (daniel-goldstein) [reply infra/gcp/README.md:45]: This is already prefixed by the cloud (`gcp`), do you mean we would run multiple instances of hail in the same cloud?

**Review by jigold — APPROVED** [review]

**Pull request merged**